### ISSUE TITLE
postman latest version (9.18.0) variables in the environment file are not updating .With ths same script session-id variable  is updating in the postman version 9.09 and below 

### ISSUE THREAD

**Issue by neerajashetty** [a85/POSTMan-Chrome-Extension#10905]
### Is there an existing issue for this?

- [X] I have searched the tracker for existing similar issues and I know that duplicates will be closed

### Describe the Issue

postman latest version (9.18.0) variables in the script are not updating .With the same script ,session-id is updating in the postman version 9.09 and below version . facing problem with the latest version 9.18.0 and 9.17 as well . 

variables which are configured in the environment files . this variable value is not updating properly (in consistent) using the script below , **the same below script which is working for 9.09 version .**

```
var data = postman.getResponseHeader("Location")
if(responseCode.code === 302) {
    var res = data.split("?");
    var loc = res[1].split("=");
    postman.setEnvironmentVariable("ob-session", loc[1]);
}
pm.test(" Authorize", function () {
    pm.response.to.have.status(302);
});
```
here ob-session variable value is not in sync with session id currently getting . 
ex: Location header which contains - **https://dev.hsetic.digitalapicraft.com/login?session_identifier=-N1mb3twGEEcA63**

### Steps To Reproduce

The below script which used in postman to set the ob-session . 
var data = postman.getResponseHeader("Location")
if(responseCode.code === 302) {
    var res = data.split("?");
    var loc = res[1].split("=");
    postman.setEnvironmentVariable("ob-session", loc[1]);
}
pm.test(" Authorize", function () {
    pm.response.to.have.status(302);
});
```

for ex : Location header which contains - **https://dev.hsetic.digitalapicraft.com/login?session_identifier=-N1mb3twGjzJqFE63**
the above script will split and set ob-session with the value N1mb3twGjzJqFEEcA63. **but this session_id is not updating in the environment variable **ob-session** that we set in the environment file** 
in the below screen- shot 




 the ob-session variable is configured in the environment variable 
![Screenshot from 2022-05-11 16-24-48](https://user-images.githubusercontent.com/88793149/167833529-25d59a1c-c583-4149-933d-0e6e5b289812.png)

Error msg - the session-id that we are getting is not populating in the environment variables ob-session . 
![Screenshot from 2022-05-11 16-30-24](https://user-images.githubusercontent.com/88793149/167834636-74d3a8c5-410f-497f-93c0-09be4fae6fe3.png)





### Screenshots or Videos


![Screenshot from 2022-05-11 16-30-24](https://user-images.githubusercontent.com/88793149/167834986-62d93f53-fcd7-4a14-8edb-caf97ddbfbc8.png)

![Screenshot from 2022-05-11 16-27-37](https://user-images.githubusercontent.com/88793149/167834149-352005e4-599b-4af1-b60f-e1f4fa1cefbe.png)
 -ob-session variable value we are getting is not in compliance with the  session id .
environment variable -
![Screenshot from 2022-05-11 16-24-48](https://user-images.githubusercontent.com/88793149/167833529-25d59a1c-c583-4149-933d-0e6e5b289812.png)

the script to set the ob-session variable 
![Script](https://user-images.githubusercontent.com/88793149/167829603-a05af710-0eea-4c61-bd6f-0c6c96d7279d.png)




### Operating System

LINUX/MAC

### Postman Version

9.18.0

### Postman Platform

Postman App

### Additional Context?

_No response_

**Comment by pmalecka** [comment]
Hi neerajashetty,

what is the response time for the API you are calling?

**Comment by neerajashetty** [comment]
@pmalecka

>  response time for the API is 
     733ms

**Comment by pmalecka** [comment]
@neerajashetty I am getting similar issues, when the API responds between 500ms and 1500ms. Any chance you could test this with an API that responds in 500ms or less?

**Comment by neerajashetty** [comment]
> @pmalecka There is no API with response time below 500ms

**Comment by pmalecka** [comment]
@neerajashetty any API that responds in more than 1500ms, that you could test this on?

**Comment by neerajashetty** [comment]
@pmalecka  Tested for that as well ,No  API that responds in  more than 1500ms.

**Comment by pmalecka** [comment]
We have reported a similar issue to help@postman.com, under ticket # 106078. 

In our case, the environment variables are not saved, when both pre-request script and test script contain `pm.environment.set` clause and the API takes between 1000ms and 1200ms to respond (in some cases the range is greater). This looks like a timing issue/race condition in Postman itself.

**Comment by timbochamp** [comment]
Thanks, @pmalecka. I'll add a bug here and link it to that support ticket/escalation!

**Comment by vdcardenas** [comment]
I have the same problem even using previously the pm.environment.unset to force the update

Postman: Version 9.18.2
Script: pm.environment.set

**Comment by vedkribhu** [comment]
@neerajashetty @pmalecka @vdcardenas  fix for this is deployed with 9.18.3(app) and on web as well. @pmalecka a special thanks for all the insights you provided, it helped us find the root cause.
